packages/zoe/src/contractSupport/zoeHelpers.js

```diff
@@ -269,3 +269,66 @@ export const assertUsesNatMath = (zcf, brand) => {
     details`issuer must use NAT amountMath`,
   );
 };
+
+/**
+ * Deposit payments such that their amounts are reallocated to a seat.
+ * The `amounts` and `payments` records must have corresponding
+ * keywords.
+ *
+ * @param {ContractFacet} zcf
+ * @param {ZCFSeat} recipientSeat
+ * @param {AmountKeywordRecord} amounts
+ * @param {PaymentKeywordRecord} payments
+ * @returns {Promise<string>} `Deposit and reallocation successful.`
+ */
+export async function depositToSeat(zcf, recipientSeat, amounts, payments) {
+  assert(!recipientSeat.hasExited(), 'The recipientSeat was exited.');
+
+  // We will create a temporary offer to be able to escrow our payments
+  // with Zoe.
+  function reallocateAfterDeposit(tempSeat) {
+    // After the assets are deposited, reallocate them onto the recipient seat and
+    // exit the temporary seat. Note that the offerResult is the return value of this
+    // function, so this synchronous trade must happen before the
+    // offerResult resolves.
+    trade(
+      zcf,
+      { seat: tempSeat, gains: {} },
+      { seat: recipientSeat, gains: amounts },
+    );
+    tempSeat.exit();
+    return `Deposit and reallocation successful.`;
```

Comment:
Move to a constant in case we want to be able to test against it

Reply:
Done